docs/CMakeLists.txt

```
@@ -13,11 +15,26 @@ set(DOXYGEN_USE_MDFILE_AS_MAINPAGE ${HOMEPAGE_MD})
 set(DOXYGEN_TIMESTAMP YES)
 set(DOXYGEN_WARN_AS_ERROR YES)
 set(DOXYGEN_SOURCE_BROWSER YES)
+set(DOXYGEN_WARN_IF_UNDOCUMENTED NO)
+
+#---- Extract files for documentation.
+get_target_property(KokkosUtils_FILES_FOR_DOC KokkosUtils INTERFACE_SOURCES)
+get_property(KOKKOSUTILS_FILES_FOR_DOC GLOBAL PROPERTY KOKKOSUTILS_FILES_FOR_DOC)
 
 #---- Add Doxygen as a target.
 #     See also https://cmake.org/cmake/help/latest/module/FindDoxygen.html.
 doxygen_add_docs(
     docs
 
+    # Homepage
     ${HOMEPAGE_MD}
+
+    # Headers from KokkosUtils library
+    ${KokkosUtils_FILES_FOR_DOC}
+
+    # Files added to the global property KOKKOSUTILS_FILES_FOR_DOC
+    ${KOKKOSUTILS_FILES_FOR_DOC}
```

Wondering if `CMake` is case sensitive... It seems so, so it might not be an "code" issue, but still it's a readability issue so I would change:
```suggestion
    # Headers from Kokkos::utils library
    $<TARGET_PROPERTY:Kokkos::utils,INTERFACE_SOURCES>

    # Files added to the global property KOKKOSUTILS_FILES_FOR_DOC
    ${KOKKOSUTILS_FILES_FOR_DOC}
```
(see https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:TARGET_PROPERTY)

From https://cmake.org/cmake/help/book/mastering-cmake/chapter/Writing%20CMakeLists%20Files.html:
> CMakeLists files use variables much like any programming language. CMake variable names are case sensitive and may only contain alphanumeric characters and underscores.

---

OK. Works like a charm :golf:

---

Actually, I had tested it only for the configure step. In the build step, it seems it doesn't work.